apps/sertra/api/PackageDataFetcher.ts

```diff
@@ -0,0 +1,49 @@
+import { ListingMod, TSListingMod, TSListingModVersion } from "./models";
+import { ApiURLs, TsApiURLs } from "./urls";
+
+interface TSListingModVersionWithOwner extends TSListingModVersion {
+  owner: string | null;
+}
+
+export const FetchListingData = async (listingId: string) => {
+  const listing_data_fetch = fetch(ApiURLs.ServerDetail(listingId)).then((x) =>
+    x.json()
+  );
+  const mod_data_fetch = fetch(TsApiURLs.V1Packages("v-rising")).then((x) =>
+    x.json()
+  );
+  const listing = await listing_data_fetch;
+  const mods = await mod_data_fetch;
+
+  const all_versions = new Map();
+  mods.map((mod: TSListingMod) => {
+    mod.versions.map((version) => {
+      const version_with_owner = version as TSListingModVersionWithOwner;
+      version_with_owner.owner = mod.owner;
+      all_versions.set(version_with_owner.full_name, version_with_owner);
+    });
+  });
```

Comment:
Here on the other hand I'd prefer that you use a for loop, as map is usually used when you need the return value (you don't here, you're just looping)